open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,736 @@
+     *
+     * @param value The object to be checked for being a valid JSONArray.
+     */
+    public static boolean isValidJSONArray(Object value) {
+        return value instanceof JSONArray;
+    }
+
+    /**
+     * Validates the Maximum Individual Amount in the control parameters of a consent request.
+     *
+     * @param controlParameters The JSON object representing the control parameters of the consent request.
+     * @return A JSON object containing the validation response.
+     */
+    public static JSONObject validateMaximumIndividualAmount(JSONObject controlParameters) {
+
+        JSONObject validationResponse = new JSONObject();
+
+        //Validate Maximum individual amount in control parameters
+        if (controlParameters.containsKey(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT)) {
+
+            Object maximumIndividualAmount = controlParameters.
+                    get(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT);
+
+            String errorMessage = String.format(ErrorConstants.INVALID_PARAMETER_MESSAGE,
+                    "maximum individual amount", "JSONObject");
+
+            // Check if the control parameter is valid
+            if (!isValidJSONObject(maximumIndividualAmount)) {
+                return ConsentManageUtil.getValidationResponse(errorMessage);
+            }
+
+            JSONObject maximumIndividualAmountResult = validateJsonObjectKey((JSONObject) maximumIndividualAmount,
+                    ConsentExtensionConstants.AMOUNT);
+            if (!(Boolean.parseBoolean(maximumIndividualAmountResult.
+                    getAsString(ConsentExtensionConstants.IS_VALID)))) {
+                return maximumIndividualAmountResult;
+            }
+        } else {
+            log.error(ErrorConstants.MISSING_MAXIMUM_INDIVIDUAL_AMOUNT);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.MISSING_MAXIMUM_INDIVIDUAL_AMOUNT);
+
+        }
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    /**
+     * Validates the Currency in Maximum Individual Amount in the control parameters of a consent request.
+     *
+     * @param controlParameters The JSON object representing the control parameters of the consent request.
+     * @return A JSON object containing the validation response.
+     */
+    public static JSONObject validateMaximumIndividualAmountCurrency(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        Object maximumIndividualAmount = controlParameters.
+                get(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT);
+
+
+        JSONObject maximumIndividualAmountResult = validateJsonObjectKey((JSONObject) maximumIndividualAmount,
+                ConsentExtensionConstants.CURRENCY);
+        if (!(Boolean.parseBoolean(maximumIndividualAmountResult.getAsString(ConsentExtensionConstants.IS_VALID)))) {
```

Comment:
Do we need to create new jsonObject here? `validationResponse` , Can't we return `maximumIndividualAmountResult` instead? wdut?